llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -2999,6 +2999,14 @@ SDValue SITargetLowering::LowerFormalArguments(
           NewArg = DAG.getMergeValues({NewArg, Chain}, DL);
         }
       } else {
+#ifndef NDEBUG
+        if (Arg.isOrigArg()) {
+          Argument *OrigArg = Fn.getArg(Arg.getOrigArgIndex());
+          assert(!OrigArg->hasAttribute("amdgpu-hidden-argument") &&
+                 "Hidden arguments should be preloaded");
```

Comment:
No error on valid IR can be an assert. This should report a proper error (via DiagnosticInfo)